plugins/auth-supertokens/src/core/addSupertokensFiles.ts

```diff
@@ -0,0 +1,244 @@
+) => {
+    const templatePath = resolve(constants.templatesPath, "supertokens.service.template.ts");
+    const template = await readFile(templatePath);
+    const templateMapping = {
+        EMAIL_FIELD_NAME: builders.identifier(settings.emailFieldName),
+        PASSWORD_FIELD_NAME: builders.identifier(settings.passwordFieldName),
+        SUPERTOKENS_ID_FIELD_NAME: builders.identifier(constants.SUPERTOKENS_ID_FIELD_NAME),
+        AUTH_ENTITY_SERVICE_ID: getAuthEntityServiceId(authEntityName),
+        AUTH_ENTITY_ID: builders.identifier(authEntityName),
+        DEFAULT_FIELD_VALUES: getDefaultCreateValues(
+            authEntityCreateInput,
+            settings.emailFieldName,
+            settings.passwordFieldName
+        )
+    }
+    appendImports(template, [
+        authEntityServiceImport(srcDirectory, authDirectory, authEntityName),
+        authEntityImport(srcDirectory, authDirectory, authEntityName)
+    ])
+    interpolate(template, templateMapping);
+    modules.set({
+        path: join(authDirectory, "supertokens", "supertokens.service.ts"),
+        code: print(template).code
+    })
+}
+
+const getDefaultCreateValues = (
+    createInput: NamedClassDeclaration,
+    emailFieldName: string,
+    passwordFieldName: string
+): namedTypes.ObjectExpression => {
+    const defaultValues: namedTypes.ObjectProperty[] = [];
+    visit(createInput, {
+        visitClassProperty: function(path) {
+            const prop = path.node;
+            //@ts-ignore
+            const propName = prop.key.name;
+            if(propName !== SUPERTOKENS_ID_FIELD_NAME && propName !== emailFieldName
+                //@ts-ignore
+                && propName !== passwordFieldName && !prop.optional) {
+                    if(!prop.typeAnnotation) {
+                        throw new Error("Failed to find the type annotation of a property");
+                    }
+                    const propType = prop.typeAnnotation.typeAnnotation;
+                    if(!propType || !propType.type) {
+                        throw new Error(`Failed to find the type annotation of the auth entity create input property: ${propName}`)
+                    }
+                    if(propName === "roles") {
+                        defaultValues.push(builders.objectProperty(
+                            builders.identifier("roles"),
+                            builders.arrayExpression([])
+                        ));
+                    } else {
+                        defaultValues.push(builders.objectProperty(
+                            builders.identifier(propName),
+                            //@ts-ignore
+                            getDefaultValueForType(propType)
+                        ))
+                    }
+            }
+            return false;
+        }
+    })
+    return builders.objectExpression(defaultValues);
+}
+
+const getDefaultValueForType = (propType: namedTypes.TSTypeAnnotation["typeAnnotation"]): any => {
```

Comment:
Try to compartmentalize the code in order to increase readability and maintainability.

Reply:
@overbit, please what exactly do you mean by compartmentalize?